[stable/datadog] update chart to support agent6 beta

xvello opened the pull request:
- upgrade kube-state-metrics dependency
- use `DD_` prefixed envvars when supported by agent5
- add commented lines for using agent6 beta images instead of agent5
- cleanup pod specs: remove empty envvars and volumes
- update readme

Except for the updated k-s-m, this PR should not affect current agent5 deployments, but allows agent6 beta users to deploy with this chart.

Also, added a fix for #2930, as we didn't refer to the right variable name for resource limits.

xvello committed:
Update Datadog chart to prepare for agent6

- upgrade kube-state-metrics dependency
- use DD_ prefixed envvars when supported by agent5
- add commented lines for using agent6 images
- cleanup pod specs: remove empty envvars and volumes
- remove kubernetes_state.yaml autoconf as it's shipped by default
- update readme

xvello committed:
fix ressources in spec, fixes #2930

xvello committed:
linting

hkaj commented on stable/datadog/requirements.yaml:
```diff
@@ -1,5 +1,5 @@
 dependencies:
   - name: kube-state-metrics
-    version: 0.3.1
+    version: ~0.5.2
```
what's the `~` for?

xvello replied:
This uses a [patchlevel match](https://docs.helm.sh/chart_best_practices/#requirements-files) to allow `>= 0.5.2 && < 0.6.0`. As the kube-state helm is pretty stable and reliable we should allow users to use a newer version when released without waiting on us to update the chart

hkaj reviewed (commented):
looking good, i'm only worried about the impact of changing the defaults in values.yaml. Can you check the upgrade goes for people who don't modify values.yaml?

xvello commented:
@hkaj thanks for the review

The only change in `values.yaml` is removing the kube-state autoconf template which was already shipped in 5.12 9 month ago. I removed it because, although it's just overriding the shipped one in agent5, it creates a duplicate in agent6 as we reworked the conf.d hierarchy.

I'll take some time to test updates before merging

CharlyF commented:
This is great! Also, by bumping the ksm version, people will start collecting many more metrics.

unguiculus commented:
/ok-to-test

unguiculus commented:
/lgtm
